engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/ClassUtils.java

```diff
@@ -116,4 +122,26 @@ public static SyntheticRepository getRepository(String classPath) {
         }
         return result;
     }
+
+    static boolean foundReflectionInMethodCalls(ConstantPool obj) {
+        return Arrays.asList(obj.getConstantPool()).parallelStream().filter(Objects::nonNull).filter(c -> c.getTag() == Const.CONSTANT_Methodref).
+                anyMatch(c -> {
+                        String invokedClassName = getClassNameFromMethodCall(obj, (ConstantMethodref) c);
+                        return reflectionClassSearchPatters.parallelStream().anyMatch(p -> invokedClassName.contains(p));
+                });
+    }
+
+    static String getClassNameFromMethodCall(ConstantPool obj, ConstantMethodref methodRef) {
+        try {
+            int referencedClassIdx = methodRef.getClassIndex();
+            int referencedClassNameIdx = ((ConstantClass) obj.getConstant(referencedClassIdx)).getNameIndex();
+            return getConstantUTF8Bytes(obj, referencedClassNameIdx);
+        } catch (Exception e) {
+            return ""; // don't fail the process
```

Comment:
log.error? log.warnning?

Reply:
debug maybe. this will most probably won't happen, and if it will then it's a minor issue.

Reply:
In which scenario you would have an Exception? Seems to me that if happens, then is not a minor issue but a system error that should be propagated as a RuntimeException and fail the process.

Reply:
ok. I can't think of a valid scenario. I'll change this